src/rust/sdk/src/tx_builder.rs

```diff
@@ -32,11 +36,19 @@ impl<'a> EconiaTransactionBuilder<'a> {
         }
     }
 
+    /// Set the retry amount for the transaction.
+    ///
+    /// Arguments:
+    /// * `amount`: the amount of times to retry the transaction.
     pub fn retry_amount(mut self, amount: u8) -> Self {
         self.retry_amount = Some(amount);
         self
     }
 
+    /// Set the internal [EntryFunction] as register_for_coin
+    ///
+    /// Arguments:
+    /// * `amount`: the amount of times to retry the transaction.
     pub fn register_for_coin(mut self, coin: &TypeTag) -> Self {
         let managed_coin = "0x1::managed_coin";
```

Comment:
Why this instead of the standard `coin.move` module?

Reply:
managed_coin does automatic registration and adds all the event handlers, i think its easier?